sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
> I think the result will be ('2015-09-20', '2015-09-20 20:00:00').

Not if we make the proposed change to force UTC for dates (and if we don't make that change then we've just moved which hours of the day will have incorrect results).

> in other words, I think the current implementation is correct, and that someInterval::date is a rounding operation which always strips hour, minute, second.

Not as long as the server and client are working in different timezones. The server sets hours, minutes and seconds to zero, then when the client tries to print a local date it applies the timezone adjustment and discards hours again.